packages/flutter/test/widgets/slivers_evil_test.dart

```diff
@@ -194,4 +194,61 @@ void main() {
     await tester.pumpAndSettle(const Duration(milliseconds: 122));
 
   });
+
+  testWidgets('Removing offscreen items does not crash', (WidgetTester tester) async {
+    await tester.pumpWidget(new MaterialApp(
+      home: new CustomScrollView(
+        slivers: <Widget>[
+          new SliverFixedExtentList(
+            itemExtent: 100.0,
+            delegate: new SliverChildBuilderDelegate(
+              (BuildContext context, int index) {
+                return new Container(
+                  color: Colors.blue,
+                  child: new Text(index.toString()),
+                );
+              },
+              childCount: 30,
+            ),
+          ),
+        ],
+      ),
+    ));
+
+    await tester.drag(find.text('5'), const Offset(0.0, -500.0));
+    await tester.pump();
+
+    // Screen is 600px high. Moved bottom item 500px up. It's now at the top.
+    expect(tester.getTopLeft(find.widgetWithText(DecoratedBox, '5')).dy, 0.0);
+
+    // Stop returning the first 2 items.
+    await tester.pumpWidget(new MaterialApp(
+      home: new CustomScrollView(
+        slivers: <Widget>[
+          new SliverFixedExtentList(
+            itemExtent: 100.0,
+            delegate: new SliverChildBuilderDelegate(
+              (BuildContext context, int index) {
+                if (index > 3) {
+                  return new Container(
+                    color: Colors.blue,
+                    child: new Text(index.toString()),
+                  );
+                }
+                return null;
+              },
+              childCount: 30,
+            ),
+          ),
+        ],
+      ),
+    ));
+
+    await tester.drag(find.text('5'), const Offset(0.0, 400.0));
+    await tester.pump();
+
+    // Move up by 4 items, meaning item 1 would have been at the top but
+    // 0 through 3 no longer exist, so item 4, 3 items down, is the first one.
+    expect(tester.getTopLeft(find.widgetWithText(DecoratedBox, '4')).dy, 300.0);
```

Comment:
This probably tests a behaviour that's not what we wanted.

Reply:
yeah, dy should be 0.0 here I think.

Reply:
it did I think https://github.com/flutter/flutter/pull/12919/files#diff-42ddda4bce1f6a6db3e1ffadcd31365fR254

Reply:
oh never mind i had incorrectly remembered that this was a diff to an existing test instead of a new test.